File: flutter_trainer_app/lib/src/features/trainer/trainer_verification_requirements.dart
List<String> missingTrainerVerificationRequirements({
  required String bio,
  required List<String> specializations,
  required String experienceYears,
  required List<Map<String, dynamic>> certifications,
}) {
  final missing = <String>[];

  if (bio.trim().isEmpty) {
    missing.add('professional bio');
  }
  if (!specializations.any((value) => value.trim().isNotEmpty)) {
    missing.add('specialization');
  }
  if (experienceYears.trim().isEmpty) {
    missing.add('years of experience');
  }
  if (!certifications.any(_hasCertificationEvidence)) {
    missing.add('certification or qualification');
  }

  return missing;
}

bool _hasCertificationEvidence(Map<String, dynamic> certification) {
  final name = certification['name']?.toString().trim() ?? '';
  final fileUrl = certification['file_url']?.toString().trim() ?? '';
  return name.isNotEmpty || fileUrl.isNotEmpty;
}

File: backend_laravel/app/Support/TrainerVerificationRequirements.php
<?php

namespace App\Support;

use App\Models\TrainerProfile;

final class TrainerVerificationRequirements
{
    /**
     * @return array<string, string>
     */
    public static function missing(TrainerProfile $profile): array
    {
        $missing = [];

        if (blank($profile->bio)) {
            $missing['bio'] = 'Add a professional bio before verification can be approved.';
        }
        if (collect($profile->specializations ?? [])->filter(fn ($value) => filled($value))->isEmpty()) {
            $missing['specializations'] = 'Add at least one specialization before verification can be approved.';
        }
        if ($profile->experience_years === null) {
            $missing['experience_years'] = 'Add your years of experience before verification can be approved.';
        }
        if (collect($profile->certifications ?? [])->filter(function ($certificate): bool {
            return is_array($certificate)
                ? filled($certificate['name'] ?? null) || filled($certificate['file_url'] ?? null)
                : filled($certificate);
        })->isEmpty()) {
            $missing['certifications'] = 'Add at least one certification or qualification before verification can be approved.';
        }

        return $missing;
    }
}
